### DISCUSSION TITLE
fieldmap and localfieldmap

### DISCUSSION THREAD

**Discussion by linsi25** [kschan0214/sepia#62] [Q&A]
Hy,

I am new to QSM and I am using SEPIA for generating my first chi-maps.  It looks quite great! 
Anyway, I have scrolled through out the documentation but I am not able to find how the fieldmap is generated after the phase unwrapping, could you help me? 
I have both brain and phantom images. The phantom is a cylinder (24 cm diameter) filled with water with 4 inserts (about 6 cm diameter) with different susceptibility values. While SEPIA seems to function properly on brain images, I have unsatisfactory results for the phantom: in particular the phantom’s localfieldmap appears quite flat (I have tried different algorithms but nothing works). Have you any ideas about why in this simpler geometry the algorithm fails?
Is there any method to check the ‘goodness’ of the localfieldmap generated by the background removal step?
Thank you very much for your attention.

**Comment by kschan0214** [comment]
Hi, 

The quality of the output fieldmap and local field map depends on several factors, including the methods being used and the signal mask. For details about how the fieldmap is generated, you can check the reference listed here https://sepia-documentation.readthedocs.io/en/latest/pages/References.html#phase-unwrapping for the methods that you choose. 

I would need more information to understand why the background field removal algorithms don't work on the phantom data (It worked on my phantom data in the past). Would you mind providing the following information if possible:
- a picture of the mask image used for the input with overlaying on the magnitude image for the phantom data (one slice including the inserts would be sufficient)
- a picture of the fieldmap of the same slice (including the display range)
- a picture of the local field of the same slice (including the display range)
- the sepia_config.m file that contains the selected algorithm parameters

I do not know any of the algorithms that provide the goodness of fit measure. One way to check the local field map output is to look at the removed background field. This can be done by subtracting the fieldmap from the local field map: 
```
Background field = fieldmap - local field
```
In principle, the derived background field should not contain any identifiable structures of the inserts.

**Reply by linsi25** [reply]
Thank you very much for you answer.
Here I attached the pictures relative to our phantom and the config file you asked for.
Hope you could understand why the algorti s do not work with this phantom.
Actually, we tried with a new phantom with smaller inserts (from 6 to 1 cm diameter), so we had more medium with ‘known’  suscrtptibiity (chi_zero) all around, and we had good results. 
Should the phantom meet any requirements? What are they? 
![mag_mask_slice_84](https://user-images.githubusercontent.com/112083858/228835211-adf50ccb-658f-4ba5-b1c9-751fe5f12c55.jpg)
![fieldmap_slice_84](https://user-images.githubusercontent.com/112083858/228835339-54f4e621-774a-4611-89b7-b8b4f7f4addb.jpg)
![localfieldmap_slice_84](https://user-images.githubusercontent.com/112083858/228836269-88fa612d-4f6e-48aa-ac95-2c80b5616f7f.jpg)

[sepia_config_1.txt](https://github.com/kschan0214/sepia/files/11111461/sepia_config_1.txt)
